PlayList.java

```diff
@@ -71,34 +100,83 @@ public int indexOf(String title) {
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
     public boolean add(int i, Track track) {
-        //// replace the following statement with your code
-        return false;
+    
+        if(size==maxSize || i<0 || i>=size) //if list is full or the given index is negative or bigger than size
+        {
+            return false;
+        }
+
+        if((tracks==null || size==0) && maxSize!=size)
+        {
+            tracks[i]=track;
+            
+        }
```

Comment:
1. Even though you could ignore cases that tracks in null - it's important to understand the meaning of null.
Here, if tracks is null you might enter the scope of the condition where you try to approach item of null object which will throw an error since null is not array. 
Most of times if we need some object value when it null, we probably use some fix logic (like initializing the array now) or throw error.
2. When maxSize = 0 this condition will be false so it will enter the else which leads to nothing be done but still return true.
3. Try to write your code more intuitive so it become more easy to the reader to understand - your conditions here make no direct sense.